internal/app/machined/pkg/controllers/network/route_spec.go

```diff
@@ -203,7 +203,9 @@ func (ctrl *RouteSpecController) syncRoute(ctx context.Context, r controller.Run
 				existing.Protocol == uint8(route.TypedSpec().Protocol) &&
 				existing.Attributes.OutIface == linkIndex && existing.Attributes.Priority == route.TypedSpec().Priority &&
 				(value.IsZero(route.TypedSpec().Source) ||
-					existing.Attributes.Src.Equal(route.TypedSpec().Source.AsSlice())) {
+					existing.Attributes.Src.Equal(route.TypedSpec().Source.AsSlice())) &&
+				(value.IsZero(route.TypedSpec().MTU) ||
```

Comment:
`value.IsZero` is good when you have structures which, because of Go parser, you cannot compare using `==`. That is - you cannot do `if myVal == MyStruct{} {` but for some reason you can do `return myval == MyStruct{}` or assign the result.

For simple type like ints, strings and others I think it is better to check using `==`.